parser/src/java/org/apache/hadoop/hive/ql/parse/HiveParser.g

```diff
@@ -1942,9 +1942,8 @@ metastoreCheck
 @after { popMsg(state); }
     : KW_MSCK (repair=KW_REPAIR)?
       (KW_TABLE tableName
-        ((add=KW_ADD | drop=KW_DROP | sync=KW_SYNC) (parts=KW_PARTITIONS))? |
-        (partitionSpec)?)
-    -> ^(TOK_MSCK $repair? tableName? $add? $drop? $sync? (partitionSpec*)?)
+        ((add=KW_ADD | drop=KW_DROP | sync=KW_SYNC) (parts=KW_PARTITIONS) (filterPartitionSpec)?)?)
+    -> ^(TOK_MSCK $repair? tableName? $add? $drop? $sync? (filterPartitionSpec)?)
```

Comment:
I know it was here before - but let's fix this up:

instead of separate add/drop/sync variable ...we could have `opt=(KW_ADD|KW_DROP|KW_SYNC)` ? that will make the other end more readable as well

Reply:
Yes It makes sense!. Fixed.